File: solutions-blind/python/003-luhn/solution.py
def luhn_valid(digits: str) -> bool:
    s = "".join(ch for ch in digits if ch not in " -")
    if not s or not s.isdigit():
        return False
    total = 0
    alt = False
    for ch in reversed(s):
        d = ord(ch) - 48
        if alt:
            d *= 2
            if d > 9:
                d -= 9
        total += d
        alt = not alt
    return total % 10 == 0

File: solutions-blind/javascript/003-luhn/solution.mjs
export function luhn_valid(digits) {
  const s = String(digits).replace(/[ -]/g, "");
  if (!s || /\D/.test(s)) return false;
  let sum = 0;
  let alt = false;
  for (let i = s.length - 1; i >= 0; i--) {
    let d = s.charCodeAt(i) - 48;
    if (alt) {
      d *= 2;
      if (d > 9) d -= 9;
    }
    sum += d;
    alt = !alt;
  }
  return sum % 10 === 0;
}
